compiler/filedescriptor.go

```diff
@@ -0,0 +1,200 @@
+package compiler
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/protobuf/parser"
+	pb "google.golang.org/protobuf/types/descriptorpb"
+)
+
+type messageBuilder struct {
+	proto3      bool
+	messageDesc *pb.DescriptorProto
+	types       types
+	scope       []string
+}
+
+func NewFileDescriptor(p *Proto, types types) *pb.FileDescriptorProto {
```

Comment:
This public function shouldn't accept a private type as a parameter. Either make the type public or the function private.

Reply:
I'll make it private for now.